### PR TITLE
Default config should type LogLevel as an enum, not a string

### LINKED ISSUE #956: [dagit] execute does not stop on certain errors
Repro:
Run against intro_tutorials
Set log_level in the config yaml to an invalid value
Execute
See error in console but Dagit stays in running state

### PR DESCRIPTION
**Pull request by bengotow**
This PR resolves #956 by changing the default environment config to use an enum for log_level. Previously log_level was a freeform string in the config schema and we'd get all the way to building the environment before it'd crash with a KeyError trying to look up the numeric log level.

This also fixes the typeahead and makes Dagit warn you before you click Run that the config is invalid:

image

### CODE REVIEW

**Comment by bengotow** [comment]
Hmm it appears the python-side config scaffolding does not support scaffolding enums. Will fix that I guess!

**Comment by mgasner** [comment]
what if we scaffolded enums as `'VALUE_1|VALUE_2|VALUE_3...'`

File: python_modules/dagster/dagster/utils/logging.py [hunk 1]
```diff
@@ -10,11 +10,18 @@
...
 from dagster import check, seven
+from dagster.core.types.config import Enum, EnumValue
 
 VALID_LEVELS = set([CRITICAL, DEBUG, ERROR, INFO, WARNING])
 
 LOOKUP = {'CRITICAL': CRITICAL, 'DEBUG': DEBUG, 'ERROR': ERROR, 'INFO': INFO, 'WARNING': WARNING}
 
+VALID_LEVEL_STRINGS = list(LOOKUP.keys())
+
+LogLevelEnum = Enum(
```
Comment (schrockn) [inline python_modules/dagster/dagster/utils/logging.py]: could be just this i think
```
LogLevelEnum = Enum('log_level', list(map(EnumValue, VALID_LEVEL_STRINGS)))
```

**Comment by bengotow** [comment]
> what if we scaffolded enums as `'VALUE_1|VALUE_2|VALUE_3...'`

Ahh I like this idea - I'll go ahead and make this change and merge this!

**Pull request merged**